CMakeLists.txt

```diff
@@ -469,6 +472,9 @@ if (BUILD_LIBWASM)
     target_compile_options(wasm PRIVATE $<$<COMPILE_LANGUAGE:CXX>:-Wno-old-style-cast>)
     target_compile_definitions(wasm PRIVATE "WASM_API_EXTERN=__attribute__((visibility(\"default\")))")
   endif ()
+  if (CMAKE_SYSTEM_NAME STREQUAL "WASI")
+    set_target_properties(wasm PROPERTIES LINK_FLAGS "-mexec-model=reactor")
```

Comment:
BTW, I tried filing https://github.com/llvm/llvm-project/issues/63207

Reply:
What we really want here I think is to make libwasm be a static library in this case.

Perhaps the simplest solution here would be to set `-sBUILD_LIBWASM=0`  .. otherwise we could need to somehow auto-detect SHARED vs STATIC or add yet another option.. which doesn't seem worth it.

Reply:
Thanks, I disabled BUILD_LIBWASM and reverted this change - actually I noticed it seems the default archives do not contain libwasm (presumably since it'd generally be hard to get a dynamic library built on CI to work outside CI) so this shouldn't have any negative effect